src/components/WorkspaceSwitcher/WorkspaceSwitcher.tsx

```diff
@@ -0,0 +1,373 @@
+import { Panel, PanelMain, PanelMainBody } from '@patternfly/react-core/dist/dynamic/components/Panel';
+import { Tab, TabTitleText, Tabs } from '@patternfly/react-core/dist/dynamic/components/Tabs';
+import { TreeViewDataItem } from '@patternfly/react-core/dist/dynamic/components/TreeView';
+import { useFlag } from '@unleash/proxy-client-react';
+import axios, { AxiosResponse } from 'axios';
+import { atom, useAtom, useAtomValue, useSetAtom } from 'jotai';
+import * as React from 'react';
+import {
+  fetchedWorkspaces,
+  isFecthingRecentlyUsedWorkspaces,
+  isFecthingRecentlyUsedWorkspacesError,
+  isFetchingWorkspacesFromRBAC,
+  isFetchingWorkspacesFromRBACError,
+  isWorkspacesMenuExpanded,
+  recentlyUsedWorkspaces,
+  selectedWorkspace,
+  workspaceTree,
+} from '../../state/atoms/workspacesAtom';
+import RecentlySavedWorkspacesResponse from './chrome/RecentlyUsedWorkspacesResponse';
+import RBACListWorkspacesResponse from './rbac/RBACListWorkspacesResponse';
+import RecentlyUsedWorkspace from './RecentlyUsedWorkspace';
+import RecentWorkspacesComponent from './RecentWorkspacesComponent';
+import FindPathForWorkspaces from './RecentWorkspacesManager';
+import { TreeViewWorkspaceItem, instanceOfTreeViewWorkspaceItem } from './TreeViewWorkspaceItem';
+import Workspace from './Workspace';
+import WorkspaceMenuToggle from './WorkspaceMenuToggle';
+import WorkspaceTreeView from './WorkspaceTreeView';
+import BuildWorkspaceTree from './WorskpaceTreeBuilder';
+
+/**
+ * Defines the statuses we want to update in an atomic fashion when fetching
+ * the workspaces from RBAC.
+ */
+interface CombinedRBACFetchingStatus {
+  isCurrentlyFetchingWorkspacesFromRBAC: boolean;
+  didFetchingWorkspacesFromRBACFail: boolean;
+}
+
+/**
+ * An atom that helps updating the statuses in an atomic fashion when fetching
+ * the worksapces from RBAC.
+ */
+const RBACRestApiStatusWriteAtom = atom(null, (_, set, arg: CombinedRBACFetchingStatus) => {
+  set(isFetchingWorkspacesFromRBAC, arg.isCurrentlyFetchingWorkspacesFromRBAC);
+  set(isFetchingWorkspacesFromRBACError, arg.didFetchingWorkspacesFromRBACFail);
+});
+
+/**
+ * Fetches the principal's workspaces from RBAC.
+ * @returns the promise of the HTTP call.
+ */
+const fetchWorkspacesFromRBAC = (): Promise<AxiosResponse<RBACListWorkspacesResponse>> => {
+  return axios.get<RBACListWorkspacesResponse>('/api/rbac/v2/workspaces/');
+};
+
+/**
+ * Defines the statuses we want to update in an atomic fashion when fecthing
+ * the recently used workspaces from Chrome.
+ */
+interface CombinedChromeFetchingStatus {
+  isCurrentlyFetchingRecentlyUsedWorkspacesFromChrome: boolean;
+  didFetchingRecentlyUsedWorkspacesFromChromeFail: boolean;
+}
+
+/**
+ * An atom that helps updating the statuses in an atomic fashion when fetching
+ * the recently used workspaces from Chrome.
+ */
+const ChromeRestApiStatusWriteAtom = atom(null, (_, set, arg: CombinedChromeFetchingStatus) => {
```

Comment:
nit: should not have first letter uppercase

Reply:
Refactored!